### PR TITLE
chore: update ci and docker

### PR DESCRIPTION
**Pull request by umut-sahin**

### CODE REVIEW

File: .github/workflows/continuous-integration.yaml [hunk 1]
```diff
@@ -689,19 +689,12 @@ jobs:
...
           -u __token__ -p ${{ secrets.PYPI_BOT_TOKEN }} \
           -r pypi "${{ env.ARTIFACTS_PACKAGED_DIR }}"/*.whl
 
-      - name: Push package to Internal PyPi
-        if: ${{ success() && !cancelled() && fromJSON(env.IS_PRERELEASE) }}
-        run: |
-          poetry run twine upload \
-          -u "${{ secrets.INTERNAL_PYPI_BOT_USERNAME }}" -p "${{ secrets.INTERNAL_PYPI_BOT_PASSWORD }}" \
-          --repository-url "${{ secrets.INTERNAL_PYPI_URL }}" "${{ env.ARTIFACTS_PACKAGED_DIR }}"/*.whl
-
```
Comment (IceTDrinker) [inline .github/workflows/continuous-integration.yaml:704]: this means pre releases are going to be pushed to public PyPi not sure this is ok ?

Comment (umut-sahin) [reply .github/workflows/continuous-integration.yaml:704]: Development is now public anyway, and releases are done on GitHub, which means wheels will be public as well. I don't see why pushing those wheels to PyPI would be a problem. Pip can distinguish between releases and pre-releases btw, so when users do `pip install concrete-numpy`, they won't get an rc. They'll only get it if they request it explicitly.

Lastly, after 1.0, there won't be pre-releases anyway. It'll all be a minor or bugfix version bump. Maybe we should verify with @aquint-zama though.

Comment (IceTDrinker) [reply .github/workflows/continuous-integration.yaml:704]: I know PyPi and pip know how to make the difference between rc and non rc releases, my qusetion was mostly IP-wise

Comment (aquint-zama) [reply .github/workflows/continuous-integration.yaml:704]: Pushing Rc to pypi is ok as now everything is already on main/public. Maybe we should still keep a way to easily package a private branch but not sure we use it that much, wdyt?

**Review by IceTDrinker — APPROVED** [review]
Thanks, I let you judge if you need a way to release a private branch

File: .github/workflows/continuous-integration.yaml (hunk 1 above)
Comment (umut-sahin) [reply .github/workflows/continuous-integration.yaml:704]: I don't think we'd use it at all, except for very very rare occasions. And for those, we can always generate the whl manually and use it.

Comment (umut-sahin) [reply .github/workflows/continuous-integration.yaml:704]: By uploading to the internal repo manually I mean. In the end, it'll certainly take less amount of time, compared to trying to automate the process for such a rare occasion.

**Pull request merged**